src/chkstat.c

```diff
@@ -441,6 +443,86 @@ usage(int x)
   exit(x);
 }
 
+static bool
+check_have_proc(void)
+{
+  char *override = secure_getenv("CHKSTAT_PRETEND_NO_PROC");
+
+  struct statfs proc;
+  int r = statfs("/proc", &proc);
+  return override == NULL && r == 0 && proc.f_type == PROC_SUPER_MAGIC;
+}
+
+static const char proc_mount_path_pattern[] = "/tmp/chkstat.proc.XXXXXX";
```

Comment:
Would it be possible to use regular /proc for this? Maybe after an `unshare(CLONE_NEWNS)`?

Reply:
Regular `/proc` without `unshare` is unwanted, because apparently Kiwi had issues in the past with programs in their `chroot` touching stuff outside of it. And while it seems unlikely to me that anything else would be running concurrently to `chkstat` in that context, I don't want to cause any more regressions. (Especially since this approach is already so hacky and unexpected!)

A custom mount namespace should work, I guess. (Man-pages suggest this should even work back in SLE 11). In the only circumstances I can think of where that fails (containers), the `mount` fails too.

I don't care much which we way here - how much do you prefer that approach?

Reply:
So I guess we just want to protect against accidental messing around with /proc, not with rogue processes trying to do damage.

The current approach with the temporary directory for mounting proc in theory opens up a kind of DoS vector, because after the mount the proc directory will have 0666 permissions (is there any way to avoid this?), thus another (unprivileged) process could `chdir()` into the proc file system and prevent a future unmount.

I'm don't strongly favor the `unshare()` approach but it could end up to be cleaner and more importantly allow simpler code, because proc paths would be in fixed locations again.

Reply:
The code turns out rather more complex, see a9ded09061f6dcc3d189d640d689daa0dea7e414. I think I'd rather stay with the path in `/tmp`.

We should probably add the `MNT_DETACH` flag to make sure the directory gets unmounted at some point. If we also care about the remaining directory in `/tmp`, we could mount things in a subdirectory of our `mkdtemp` directory.

Reply:
I just commited e65dda3a8b26db4fbab93040acdc629b4ee41ac7 in this vein. This drives up the complexity of the `mkdtemp` option to the point where it's a wash in my book.

I guess that makes the `unshare` option the better one since it should actually guarantee the `umount` can never fail. If you agree, I'll commit that one to finally close this PR off.

Reply:
I see, both approaches have their difficulties. Okay then let's take the `unshare` route. Just ping me when you're finished you changes and I'll have a final look.